utils/logalloc.cc

```diff
@@ -1074,20 +1074,20 @@ void segment_pool::prime(size_t available_memory, size_t min_free_memory) {
     reclaim_segments(_store.non_lsa_reserve / segment::size, is_preemptible::no);
 }
 
-void segment_pool::on_segment_compaction(size_t used_size) {
+inline void segment_pool::on_segment_compaction(size_t used_size) noexcept {
```

Comment:
If we're inlining functions in a .cc file, can we just move the definition to the class? The language fluff is particularly visible on oneliners:

```
inline void on_memory_allocation(size_t size) noexcept;
inline void segment_pool::on_memory_allocation(size_t size) noexcept {
    _stats.memory_allocated += size;
}
```
vs 

`inline void on_memory_allocation(size_t size) noexcept { _stats.memory_allocated += size; }`